### PR TITLE
Support external import paths in gRPC tool

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1612
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/tests/resources/generated-sources/tool_test_external_imports/child_pb.bal [hunk 1]
```diff
@@ -0,0 +1,272 @@
+import ballerina/grpc;
```
Comment (lahirusamith) [inline ballerina/tests/resources/generated-sources/tool_test_external_imports/child_pb.bal:1]: Is licence block needed in test files? if so licence block is missing.

File: ballerina/tests/resources/generated-sources/tool_test_external_imports/parent_pb.bal [hunk 2]
```diff
@@ -0,0 +1,239 @@
+import ballerina/grpc;
```
Comment (lahirusamith) [inline ballerina/tests/resources/generated-sources/tool_test_external_imports/parent_pb.bal:1]: licence block is missing here also.

File: ballerina/tests/resources/generated-sources/tool_test_external_imports/child_pb.bal (hunk 1 above)
Comment (MadhukaHarith92) [reply ballerina/tests/resources/generated-sources/tool_test_external_imports/child_pb.bal:1]: These are test resource files that are used to asset the content of the bal files generated from the gRPC tool. Therefore they don't have license headers.

File: ballerina/tests/resources/generated-sources/tool_test_external_imports/parent_pb.bal (hunk 2 above)
Comment (MadhukaHarith92) [reply ballerina/tests/resources/generated-sources/tool_test_external_imports/parent_pb.bal:1]: These are test resource files that are used to asset the content of the bal files generated from the gRPC tool. Therefore they don't have license headers.

File: ballerina/tests/resources/proto-files/external-imports/myproj/foo/bar/child.proto [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+
```
Comment (BuddhiWathsala) [inline ballerina/tests/resources/proto-files/external-imports/myproj/foo/bar/child.proto:19]:
```suggestion

```

File: ballerina/tests/resources/proto-files/external-imports/myproj/parent.proto [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+
```
Comment (BuddhiWathsala) [inline ballerina/tests/resources/proto-files/external-imports/myproj/parent.proto:19]:
```suggestion

```
Unnecessary space.

File: native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java [hunk 5]
```diff
@@ -233,8 +261,19 @@ private void generateBalFile(String protoPath) {
             }
             LOG.debug("Successfully generated root descriptor.");
             try {
-                dependant = DescriptorsGenerator.generateDependentDescriptor(this.protocExePath,
-                        getAbsolutePath(protoPath), descFilePath.toAbsolutePath().toString());
+                if (importPath.equals("")) {
```
Comment (BuddhiWathsala) [inline native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java:264]: How about using `importPath.isBlank()` here?

Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java]: Fixed in 3f03c2b

File: ballerina/tests/resources/proto-files/external-imports/myproj/foo/bar/child.proto (hunk 3 above)
Comment (MadhukaHarith92) [reply ballerina/tests/resources/proto-files/external-imports/myproj/foo/bar/child.proto]: Fixed in 3f03c2b

File: ballerina/tests/resources/proto-files/external-imports/myproj/parent.proto (hunk 4 above)
Comment (MadhukaHarith92) [reply ballerina/tests/resources/proto-files/external-imports/myproj/parent.proto]: Fixed in 3f03c2b

**Review by BuddhiWathsala — APPROVED** [review]

**Review by lahirusamith — APPROVED** [review]

**Pull request merged**